derp/derp_server.go

```diff
@@ -45,6 +45,15 @@ var debug, _ = strconv.ParseBool(os.Getenv("DERP_DEBUG_LOGS"))
 // verbosely log whenever DERP drops a packet.
 var verboseDropKeys = map[key.Public]bool{}
 
+// Stores empty content byte buffers after they've been used, so further allocations
+// do not need to allocate more
+var bytesPool = &sync.Pool{
+	New: func() interface{} {
```

Comment:
this New func is just unnecessarily allocating, as it'll never satisfy the caller's needs.

You can just omit it entirely.  The only value it provides is documentation of the sync.Pool's return type, but you can document that instead:

```go
var bytesPool = new(sync.Pool) // of *[]byte
```